dashboard/app/controllers/levels_controller.rb

```diff
@@ -4,7 +4,7 @@
 class LevelsController < ApplicationController
   include LevelsHelper
   include ActiveSupport::Inflector
-  before_filter :authenticate_user!
+  before_filter :authenticate_user!, :unless => Proc.new {params[:embed]}
```

Comment:
oh wait, I think we only want to do the unless for the show action.

(I think the authorize! filter will catch all the cases that would get through as the code is now (except for index) but I think it's safer to limit things as much as we can (esp. if we change something later))